internal/provider/provider.go

```diff
@@ -619,7 +619,11 @@ func resourceDnsDelete(d *schema.ResourceData, meta interface{}, rrType uint16)
 
 		msg.SetUpdate(d.Get("zone").(string))
 
-		rr, _ := dns.NewRR(fmt.Sprintf("%s 0 %s", fqdn, dns.TypeToString[rrType]))
+		rr, err := dns.NewRR(fmt.Sprintf("%s 0 %s", fqdn, dns.TypeToString[rrType]))
+		if err != nil {
+			return fmt.Errorf("error reading DNS record: %s", err)
```

Comment:
Should we include the DNS record string in the error message or is it already there?

Reply:
It doesn't look to me that the error message contains the DNS record string.

In which case, are you referring to doing a nil check on `rr` and then calling `rr.String()` and adding this to the error message if it is available?

Reply:
I was thinking the available `fmt.Sprintf("%s 0 %s", fqdn, dns.TypeToString[rrType])` content